pkg/server/proto/server.proto

```diff
@@ -372,6 +372,17 @@ service Waypoint {
   // UpsertStatusReport updates or inserts a statusreport.
   rpc UpsertStatusReport(UpsertStatusReportRequest) returns (UpsertStatusReportResponse);
 
+  // Internal Task functions
+
+  // UpsertTask updates or inserts a Task
+  rpc UpsertTask(UpsertTaskRequest) returns (UpsertTaskResponse);
```

Comment:
Not fully convinced we need a write/update API for tasks at all. I think this goes back to my original feedback in the part 1 or part 2 PR. I'd manage all task state changes internally, and only expose _operations_ on tasks like `CancelTask` or something like that, rather than letting API callers modify state.

Reply:
Yep, this makes sense if we move the task driven state into the server!